neutron_policy_server/wsgi.py

```diff
@@ -16,40 +19,85 @@
 app = Flask(__name__)
 
 
-@app.route("/enforce", methods=["POST"])
-def enforce():
-    data = request.json
-    rule = data.get("rule")
-    target = data.get("target")
-    creds = data.get("creds")
-
+def _fetch_context(request):
+    content_type = request.headers.get('Content-Type', default="application/x-www-form-urlencoded")
+    if content_type == "application/x-www-form-urlencoded":
+        data = request.form.to_dict()
+    elif content_type == "application/json":
+        data = request.json
+    target = json.loads(data.get("target")) if (
+            type(data.get("target")) is str) else data.get("target")
+    creds = json.loads(data.get("credentials")) if (
+            type(data.get("credentials")) is str) else data.get("credentials")
```

Comment:
is there any reason why we're having to do a json loads if we're already breaking it up into `request.json` earlier?

Reply:
That's mostly for `application/x-www-form-urlencoded` content type

Reply:
Oh it actually does a `application/x-www-form-urlencoded` post with JSON content inside the values?! This stuff is really stinky if so :(